results/pkg/watcher/reconciler/controller_test.go

```diff
@@ -132,6 +132,7 @@ func TestReconcile(t *testing.T) {
 		tr.UID = "234435"
 		asset.Clients.Pipeline.TektonV1beta1().TaskRuns(taskRun.Namespace).Update(tr)
 		updatetr, err := reconcileTaskRun(ctx, asset, tr)
+		updatetr.ResourceVersion = tr.ResourceVersion
```

Comment:
This seems unrelated to the rest of the proto changes. Why is this necessary?

Reply:
That's because the new version of k8s will change objects' ResourceVersion after they are updated, so we need to equal them manually before comparing them. See [concurrency-control-and-consistency](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#concurrency-control-and-consistency). Should we also make sure the ResourceVersion field is changed?